docs/rackhd_vlab/web_ui.rst

```diff
@@ -1,2 +1,229 @@
+
+you can skip this option step.
+
+Below it will explain the shell commands' meaning in the last step.
+
+1. It will list the hardware by 'lshw' and catalogs and output (you can find the output from catalog after workflow completion.)
+
+.. code::
+
+    "command" : "sudo ls /var",
+    "catalog" : {
+        "format" : "raw",
+        "source" : "ls var"
+    }
+
+2. As below:This is a diagnostic sample for temperature. It's comparing the hardware's Ambient temperature with a threshold value (20 as an example) and fail this job if hotter than that.20 as an example) and fail this job if hotter than that.
+
+.. code-block:: shell
+
+    temp=$( sudo ipmitool sdr|grep Temp|head -n1| awk '{print $3}' ) && \
+    echo Temperature: $temp && \
+    if [ $temp > 20 ]; then \
+        echo [Error] Over Temperature! $temp; \
+        exit -1; \
+    fi
+
+3. It is comparing the hardware's CPU frequency with a threshold value (2500 as an example) and fail this job if lower than that.
+
+.. code-block:: shell
+
+    CPU_HZ=$(cat /proc/cpuinfo |grep MHz | head -n1 | awk '{print $4}') && \
+    echo CPU frequency : $CPU_HZ && \
+    if [ $(awk 'BEGIN{ print $CPU_HZ <2000 }') -eq 1 ]; then \
+        echo [Error] Wrong SKU. CPU frequency is too low: $CPU_HZ; \
+        exit -1; \
+    fi
+
+Set the task relationship
+---------------------------
+
+Tasks display indicators that you can connect to set the task relationship. Each task displays a
+trigger indicator in the top left.
+Each task also displays the following condition indicators on the right side:
+
+     * Red: when fail
+     * Green: when success
+     * Blue: when finish
+     * Yellow: when cancelled
+
+For example, when you connect the green condition indicator of task A to the trigger indicator for
+Task B: when task A has succeeded, then task B is triggered.
+
+Before setting the relationship we need to add waitOn input for the tasks, right click on the task block and click ``Addinput``.
+
+.. image:: ../_static/web-ui/s12.png
+    :align: center
+
+Then connect the ``finished`` output of ``set-pxe-boot`` task to ``reboot``'s ``waitOn`` input, ``reboot``'s ``succeeded`` output connect to ``bootstrap-ubuntu`` and ``diagnostic``'s ``waitOn`` input
+
+.. image:: ../_static/web-ui/s13.png
+    :align: center
+
+When the reboot task is successfully completed, the ``bootstrap-ubuntu`` task and ``diagnostic`` task are started.
+
+Now we can save the workflow, before saving the workflow we need to fill in the ``friendlyName`` and ``injectableName`` on the right of the workflow editor panel. Then click the save button.
+
+.. image:: ../_static/web-ui/s14.png
+    :align: center
+
+Go to ``Workflow Viewer`` session and filter the workflow by name. choose ``My_Workflow``
+
+.. image:: ../_static/web-ui/s15.png
+    :align: center
+
+Go to ``Run Workflow`` session, choose yout ``Node id``, ``OBM Host`` and type ``Graph.My_Workflow`` in ``Graph`` field, then click ``RUN WORKFLOW`` button
```

Comment:
Remove OBM host description.
It is used to filter nodes or display nodes details.